include/Bezier/polycurve.h

```diff
@@ -255,17 +261,23 @@ class PolyCurve
    * \return A vector af points of intersection between curves
    */
   template <typename Curve_PolyCurve>
-  std::vector<Point> pointsOfIntersection(const Curve_PolyCurve& curve, bool stop_at_first = false,
-                                             double epsilon = 0.001) const;
+  PointVector intersection(const Curve_PolyCurve& curve, bool stop_at_first = false, double epsilon = 0.001) const;
 
   /*!
    * \brief Get the parameter t where polycurve is closest to given point
    * \param point Point to project on polycurve
-   * \param step Size of step in coarse search
    * \param epsilon Precision of resulting projection
    * \return Parameter t
    */
-  double projectPoint(const Point& point, double step = 0.01, double epsilon = 0.001) const;
+  Parameter projectPoint(const Point& point, double epsilon = 0.001) const;
+
+  /*!
+   * \brief Get the parameter t vector where polycurve is closest to given points
+   * \param point_vector Points to project on polycurve
+   * \param epsilon Precision of resulting projection
+   * \return Vector of parameters t
+   */
+  ParameterVector projectPoint(PointVector point_vector, double epsilon = 0.001) const;
```

Comment:
Hm, should this be called projectPoints?

Reply:
It is intentional that it has same function name